src/azure-cli-core/azure/cli/core/_help.py

```diff
@@ -248,6 +251,12 @@ def load(self, options):
         for loader in ordered_loaders:
             loader.versioned_load(self, options)
 
+        # Used as a hook for changing the default examples
+        if self.help_ctx.original_nouns == self.command:
```

Comment:
To me, it seems the purpose of `original_nouns` is to detect whether the help is loaded for the original command like `vm` or its children, like `vm availability-set`, so that we only load AI examples for the command itself, not its children.

```
> az vm -h

Group
    az vm              << Load AI examples for it

Subgroups:
    availability-set   << Don't load AI examples for these
    boot-diagnostics
    diagnostics
...
```
If so, an easier approach is to copy the logic from `knack.help.CLIHelp.show_help` to `azure.cli.core._help.AzCliHelp.show_help` and add the replacement/addition logic here, instead of going very deep and evaluate for every `CliHelpFile.load` whether the help is retrieved for the original command - `vm`. In other words, we can actively add the AI examples after the help is rendered, instead of passively evaluating the given command.

```py
    # azure.cli.core._help.AzCliHelp.show_help
    def show_help(self, cli_name, nouns, parser, is_group):
        self.update_loaders_with_help_file_contents(nouns)
        self.original_nouns = ' '.join(nouns)
        import colorama
        colorama.init(autoreset=True)
        delimiters = ' '.join(nouns)
        help_file = self.command_help_cls(self, delimiters, parser) if not is_group \
            else self.group_help_cls(self, delimiters, parser)
        help_file.load(parser)
        if not nouns:
            help_file.command = ''
        else:
            new_examples = self.check_for_new_examples(help_file.command)  # ai examples are added here
            if new_examples:
                help_file.examples = new_examples
        self._print_detailed_help(cli_name, help_file)

        print(SURVEY_PROMPT)
```

Also, I would suggest to write some tests similar to 

https://github.com/Azure/azure-cli/blob/39ca86e9e9e98981024ed95db5167f440de1d4f6/src/azure-cli-core/azure/cli/core/tests/test_help.py#L345-L353

so that we can verify the hook is working as expected.

Reply:
Correct, that is the purpose! If that's a cleaner approach I can switch to it.

I will add test in the next PR.